Riot/Modules/Application/AppCoordinator.swift

```diff
@@ -322,15 +322,39 @@ fileprivate class AppNavigator: AppNavigatorProtocol {
         self.appCoordinator.navigate(to: destination)
     }
     
-    func addLoadingActivity() -> Activity {
-        let presenter = ActivityIndicatorToastPresenter(
-            text: VectorL10n.roomParticipantsSecurityLoading,
-            navigationController: appNavigationVC
-        )
-        let request = ActivityRequest(
-            presenter: presenter,
-            dismissal: .manual
-        )
+    func addAppActivity(_ type: AppActivityType) -> Activity {
+        let request = activityRequest(for: type)
         return ActivityCenter.shared.add(request)
     }
+    
+    // MARK: - Private
+    
+    private func activityRequest(for type: AppActivityType) -> ActivityRequest {
+        switch type {
+        case let .loading(label):
+            let presenter = ToastActivityPresenter(
+                viewState: .init(
+                    style: .loading,
+                    label: label
+                ),
+                navigationController: appNavigationVC
```

Comment:
This presenter doesn't *really* need to know that it's a navigation controller, does it?

Reply:
Actually, does it even need to be a view controller or is a plain UIView enough?